### PR TITLE
Fix/case insensitive hsts

### PR DESCRIPTION
**Pull request by noamblitz**
### Changes
According to RFC 6797:

- Directive names are case-insensitive
- Max age can contain quotes

This pr fixes that 

### Demo
_Please add some proof in the form of screenshots or screen recordings to show (off) new functionality, if there are interesting new features for end-users._

---

### Code Checklist
- [ ] All the commits in this PR are properly PGP-signed and verified.
- [ ] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [ ] I have written unit tests for the changes or fixes I made.
- [ ] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [ ] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [ ] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [ ] I have made corresponding changes to the documentation, if necessary.
- [ ] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — DISMISSED** [review]

**Review by underdarknl — APPROVED** [review]

**Comment by underdarknl** [comment]
ideally we'd just to a single call to lowercase the value.

File: octopoes/bits/check_hsts_header/check_hsts_header.py [hunk 1]
```diff
@@ -16,13 +16,13 @@ def run(input_ooi: HTTPHeader, additional_oois: List, config: Dict[str, str]) ->
     max_age = int(config.get("max-age", one_year)) if config else one_year
     findings: [str] = []
 
-    if "includeSubDomains" not in header.value:
+    if "includesubdomains" not in header.value.lower():
```
Comment (underdarknl) [inline octopoes/bits/check_hsts_header/check_hsts_header.py:19]:
```suggestion
    headervalue = header.value.lower()
    if "includesubdomains" not in headervalue:
```

File: octopoes/bits/check_hsts_header/check_hsts_header.py [hunk 2]
```diff
@@ -16,13 +16,13 @@ def run(input_ooi: HTTPHeader, additional_oois: List, config: Dict[str, str]) ->
...
         findings.append("The HSTS should include subdomains.")
 
-    if "max-age" not in header.value:
+    if "max-age" not in header.value.lower():
```
Comment (underdarknl) [inline octopoes/bits/check_hsts_header/check_hsts_header.py:22]:
```suggestion
    if "max-age" not in headervalue:
```

File: octopoes/bits/check_hsts_header/check_hsts_header.py [hunk 3]
```diff
@@ -16,13 +16,13 @@ def run(input_ooi: HTTPHeader, additional_oois: List, config: Dict[str, str]) ->
...
         findings.append("The cache validity period of the HSTS should be defined and should be at least 1 year.")
 
-    if "max-age" in header.value and int(header.value.split("=")[1].split(";")[0]) < max_age:
+    if "max-age" in header.value.lower() and int(header.value.split("=")[1].strip('"').split(";")[0]) < max_age:
```
Comment (underdarknl) [inline octopoes/bits/check_hsts_header/check_hsts_header.py:25]:
```suggestion
    if "max-age" in headervalue and int(headervalue.split("=")[1].strip('"').split(";")[0]) < max_age:
```

**Comment by stephanie0x00** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
Have verified the HSTS header for various websites, including those with and without HSTS headers. Results seem to match what the domain returns. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

**Review by underdarknl — APPROVED** [review]

**Pull request merged**